hasher-matcher-actioner/hmalib/common/writebacker_models.py

```diff
@@ -0,0 +1,240 @@
+        (e.g. "ThreatExchangeWritingBackEnabled:<collaboration-id>").
+        """
+        return True
+
+    @property
+    def te_api(self) -> ThreatExchangeAPI:
+        mock_te_api = os.environ.get("MOCK_TE_API")
+        if mock_te_api == "True":
+            return MockedThreatExchangeAPI()
+        api_key = AWSSecrets.te_api_key()
+        return ThreatExchangeAPI(api_key)
+
+
+class ThreatExchangeFalsePositiveWritebacker(ThreatExchangeWritebacker):
+    """
+    For writing back to ThreatExhcnage that the user belives the match was
+    a false positive.
+
+    Executing perform_writeback on this class will read the (indicator, privacy_group)
+    pairs for the match and, for each, add a new descriptor on that indicator
+    in that privacy group that adds the disagreement tag for the privacy group
+
+    """
+
+    def _writeback_impl(self, writeback_message: WritebackMessage) -> str:
+        # TODO Implement
+        return "Wrote Back false positive"
+
+
+class ThreatExchangeTruePositivePositiveWritebacker(ThreatExchangeWritebacker):
+    """
+    For writing back to ThreatExhcnage that the user belives the match was
+    correct.
+
+    Executing perform_writeback on this class will read the (indicator, privacy_group)
+    pairs for the match and, for each, add a new descriptor on that indicator
+    in that privacy group that adds the agreement tag for the privacy group
+
+    """
+
+    def _writeback_impl(self, writeback_message: WritebackMessage) -> str:
+        # TODO Implement
+        return "Wrote Back true positive"
+
+
+@dataclass
+class ThreatExchangeReactionWritebacker(ThreatExchangeWritebacker):
+    """
+    For writebacks to ThreatExchange that are implemented as reactions.
+
+    Executing perform_writeback on this class will read the indicators
+    from the match, load all related descriptors, and write the given reaction
+    to them
+    """
+
+    @property
+    def reaction(self) -> str:
+        raise NotImplementedError
+
+    def _writeback_impl(self, writeback_message: WritebackMessage) -> str:
+        indicator_ids = {
+            dataset_match_details.banked_content_id
+            for dataset_match_details in writeback_message.matching_banked_signals
+            if dataset_match_details.bank_source == "te"
+        }
+
+        descriptor_ids = {
+            descriptor_id["id"]
+            for indicator_id in indicator_ids
+            for descriptor_id in self.te_api.get_threat_descriptors_from_indicator(
+                indicator_id
+            )
+        }
+
+        for id in descriptor_ids:
+            self.te_api.react_to_threat_descriptor(id, self.reaction)
+            logger.info("reacted %s to descriptor %s", self.reaction, id)
+        return (
+            "reacted "
+            + self.reaction
+            + " to "
+            + str(len(descriptor_ids))
+            + " descriptors"
+        )
+
```

Comment:
Is _writeback_impl going to persist anything to be displayed in the UI? Seems like it should, like we still need in actioning.

Reply:
Could do. Dont think thats on the v0 mural though

Reply:
k